var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+from spack import *
+
+
+class Faiss(AutotoolsPackage, PythonPackage):
```

Comment:
I would also extend `CudaPackage`. This will automatically add the variant, dependency, and conflicts.